swo_aws_extension/flows/jobs/pool_notifications.py

```diff
@@ -76,48 +79,140 @@ def create_new_notification(crm_client, pls_enabled, notification_type, summary,
+
+
+def get_mpa_accounts_map():
+    """
+    Get the MPA accounts map.
+    """
+    mpa_accounts = get_mpa_accounts()
+    accounts_map = {}
+    for mpa_account in mpa_accounts:
+        if mpa_account.country not in accounts_map:
+            accounts_map[mpa_account.country] = {
+                True: 0,
+                False: 0,
+            }
+        if mpa_account.status == MPAStatusEnum.READY:
+            accounts_map[mpa_account.country][mpa_account.pls_enabled] = (
+                accounts_map[mpa_account.country][mpa_account.pls_enabled] + 1
+            )
+    return accounts_map
 
 
 def check_pool_accounts_notifications(config) -> None:
     """
     Check the pool accounts notifications.
     """
-    pls_values = [True, False]
     crm_client = get_service_client()
+    open_notifications = get_open_notifications()
+    pending_notifications = [
+        notification
+        for notification in open_notifications
+        if notification.status == NotificationStatusEnum.PENDING
+    ]
 
-    pending_notifications = get_pending_notifications()
     for pending_notification in pending_notifications:
         process_pending_notification(crm_client, pending_notification)
 
     logger.info("Check if new notifications need to be created")
-    for pls_enabled in pls_values:
-        if has_pending_notifications(pls_enabled):
-            logger.info(
-                f"Pending notification found for PLS {'enabled' if pls_enabled else 'disabled'}"
-            )
-            continue
 
-        available_mpa = get_available_mpa_from_pool(pls_enabled)
-        if not available_mpa:
-            create_new_notification(
-                crm_client, pls_enabled, NotificationTypeEnum.EMPTY, EMPTY_SUMMARY, EMPTY_TITLE
+    delete_duplicated_new_notifications(open_notifications, pending_notifications)
+    accounts_map = get_mpa_accounts_map()
+    add_new_notifications(config.minimum_mpa_threshold, accounts_map, pending_notifications)
+
+    logger.info("Proceed to create new ticket if needed")
+    new_notifications = get_notifications_by_status(NotificationStatusEnum.NEW)
+    for notification in new_notifications:
+        if notification.notification_type == NotificationTypeEnum.EMPTY:
+            ticket = create_ticket(
+                crm_client,
+                notification,
+                EMPTY_SUMMARY,
+                EMPTY_TITLE,
             )
-        elif len(available_mpa) <= config.minimum_mpa_threshold:
-            create_new_notification(
+        else:
+            ticket = create_ticket(
                 crm_client,
-                pls_enabled,
-                NotificationTypeEnum.WARNING,
+                notification,
                 NOTIFICATION_SUMMARY,
                 NOTIFICATION_TITLE,
             )
 
+        notification.status = NotificationStatusEnum.PENDING
+        notification.ticket_state = "New"
+        notification.ticket_id = ticket.get("id", "")
+        notification.save()
+
```

Comment:
Not blocking, just a suggestion for a future improvement. We could isolate this logic into a method like update_notification, passing a dict with the field name and value. That way, we can reuse it and keep the responsibility of updating notifications in one place.